src/clm/functions.py

```diff
@@ -143,13 +143,22 @@ def read_file(smiles_file, max_lines=None):
     return lines_array
 
 
-def read_file_incremental(input_file, max_lines=None):
+def read_file_incremental(
+    input_file,
+    max_lines=None,
+    smile_only=True,
```

Comment:
The purpose of introducing this flag was to not have existing code break (where `read_file` is used to read the entire line - this is why we renamed the function from `read_smiles` to `read_file`). Thus, the default value of this should be `False` (or go through the entire code to find out where it was used and change the calling code to `smile_only=False`).

Further, the caller may just use `read_file` and not `read_file_incremental`. So this flag should be passed down from `read_file` to `read_file_incremental`.